components/tools/OmeroJava/test/integration/gateway/TablesFacilityTest.java

```diff
@@ -0,0 +1,187 @@
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+package integration.gateway;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Random;
+import java.util.UUID;
+
+import omero.gateway.facility.TablesFacility;
+import omero.gateway.model.DatasetData;
+import omero.gateway.model.FileAnnotationData;
+import omero.gateway.model.TableData;
+import omero.gateway.model.TableDataColumn;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class TablesFacilityTest extends GatewayTest {
+
+    private static Random rand = new Random();
+
+    private static final int nCols = 10;
+
+    private static final int nRows = 100000;
+
+    private DatasetData ds;
+
+    private TableData original;
+
+    @Override
+    @BeforeClass(alwaysRun = true)
+    protected void setUp() throws Exception {
+        super.setUp();
+
+        DatasetData ds = new DatasetData();
+        ds.setName(UUID.randomUUID().toString());
+        this.ds = (DatasetData) datamanagerFacility.createDataset(rootCtx, ds,
+                null);
+    }
+
+    @Test(timeOut = 300000)
+    public void testAddTable() throws Exception {
+        Class<?>[] types = new Class<?>[] { String.class, Long.class,
+                Double.class, Double[].class };
+        TableDataColumn[] header = new TableDataColumn[nCols];
+        for (int i = 0; i < header.length; i++) {
+            header[i] = new TableDataColumn("column-" + i, i,
+                    types[rand.nextInt(types.length)]);
+        }
+
+        Object[][] data = new Object[header.length][nRows];
+        for (int c = 0; c < nCols; c++) {
+            Object[] column = new Object[nRows];
+            Class<?> type = header[c].getType();
+            for (int r = 0; r < nRows; r++) {
+                if (type.equals(String.class)) {
+                    column[r] = "" + rand.nextInt();
+                } else if (type.equals(Long.class)) {
+                    column[r] = rand.nextLong();
+                } else if (type.equals(Double.class)) {
+                    column[r] = rand.nextDouble();
+                } else if (type.equals(Double[].class)) {
+                    Double[] d = new Double[4];
+                    for (int i = 0; i < 4; i++)
+                        d[i] = rand.nextDouble();
+                    column[r] = d;
+                }
+            }
+            data[c] = column;
+        }
+
+        original = new TableData(header, data);
+        long start = System.currentTimeMillis();
+        TableData stored = tablesFacility.addTable(rootCtx, ds, "Big Table",
+                original);
+        long duration = System.currentTimeMillis() - start;
+
+        original.setOriginalFileId(stored.getOriginalFileId());
+
+        System.out.println("Storing table with " + (nCols * nRows)
```

Comment:
are you expecting that to be done in a given time interval?
There is a timeout parameter so it will fail if it takes too long